include/deal.II/base/symmetric_tensor.h

```diff
@@ -2320,6 +2324,401 @@ Number second_invariant (const SymmetricTensor<2,3,Number> &t)
+   * to be computed.
+   *
+   * @return An array containing the eigenvectors and the associated eigenvalues
+   *
+   * @relates SymmetricTensor
+   * @author Joachim Kopp, Jean-Paul Pelteret, 2017
+   */
+  template <int dim, typename Number>
+  std::array<std::pair<Number, Tensor<1,dim,Number> >,dim>
+  jacobi (dealii::SymmetricTensor<2,dim,Number> A);
+
+
+
+  /**
+   * Compute the eigenvalues and eigenvectors of a real-valued rank-2
+   * symmetric tensor using the characteristic equation to compute eigenvalues
+   * and an analytical approach based on the cross-product for the eigenvectors.
+   * If the computations are deemed too inaccurate then the method falls
+   * back to ql_implicit_shifts.
+   *
+   * @param[in] A The tensor of which the eigenvectors and eigenvalues are
+   * to be computed.
+   *
+   * @return An array containing the eigenvectors and the associated eigenvalues
+   *
+   * @relates SymmetricTensor
+   * @author Joachim Kopp, Jean-Paul Pelteret, 2017
+   */
+  template <typename Number>
+  std::array<std::pair<Number, Tensor<1,2,Number> >,2>
+  hybrid (const dealii::SymmetricTensor<2,2,Number> &A);
+
+
+
+  /**
+   * Compute the eigenvalues and eigenvectors of a real-valued rank-2
+   * symmetric tensor using the characteristic equation to compute eigenvalues
+   * and an analytical approach based on the cross-product for the eigenvectors.
+   * If the computations are deemed too inaccurate then the method falls
+   * back to ql_implicit_shifts.
+   * The specialized algorithm implemented here is given in
+   * Kopp, J.
+   * Efficient numerical diagonalization of hermitian 3x3 matrices
+   * International Journal of Modern Physics C, 2008, 19, 523-548
+   * doi: 10.1142/S0129183108012303
+   * arXiv.org preprint: physics/0610206
+   *
+   * @param[in] A The tensor of which the eigenvectors and eigenvalues are
+   * to be computed.
+   *
+   * @return An array containing the eigenvectors and the associated eigenvalues
+   *
+   * @relates SymmetricTensor
+   * @author Joachim Kopp, Jean-Paul Pelteret, 2017
+   */
+  template <typename Number>
+  std::array<std::pair<Number, Tensor<1,3,Number> >,3>
+  hybrid (const dealii::SymmetricTensor<2,3,Number> &A);
+
+
+
+  namespace
+  {
+    template<int dim, typename Number>
+    struct SortEigenValuesVectors
+    {
+      typedef std::pair<Number, Tensor<1,dim,Number> > EigValsVecs;
+      bool operator() (const EigValsVecs &lhs,
+                       const EigValsVecs &rhs)
+      {
+        return lhs.first > rhs.first;
+      }
+    };
+  }
+
+} // namespace internal
+
+#endif // DOXYGEN
+
+
+// The line below is to ensure that doxygen puts the full description
+// of this global enumeration into the documentation
+// See https://stackoverflow.com/a/1717984
+/** @file */
+/**
+ * An enumeration for the algorithm to be employed when performing
+ * the computation of normalized eigenvectors and their corresponding
+ * eigenvalues.
```

Comment:
... by the eigenvalues() and eigenvectors() methods operating on SymmetricTensor objects.